Modules/@babylonjs/react-native-iosandroid/android/src/main/java/com/babylonreactnative/EngineView.java

```diff
@@ -67,34 +70,58 @@ public void surfaceDestroyed(SurfaceHolder holder) {
     public void setAntiAliasing(Integer value) {
         BabylonNativeInterop.updateMSAA(value);
     }
+
+    public void setTopMost(Boolean topMost) {
+        setIsTransparentAndTopMost(this.isTransparent, topMost);
+    }
     // ------------------------------------
     // TextureView related
-
     public void setIsTransparent(Boolean isTransparent) {
+        setIsTransparentAndTopMost(isTransparent, this.topMost);
+    }
+
+    public void setIsTransparentAndTopMost(Boolean isTransparent, Boolean topMost) {
+        if (this.surfaceView != null) {
+            this.surfaceView.setVisibility(View.GONE);
+            this.surfaceView = null;
+        }
+        if (this.textureView != null) {
+            this.textureView.setVisibility(View.GONE);
+            this.textureView = null;
+        }
         if (isTransparent) {
-            if (this.opaqueSurfaceView != null) {
-                this.opaqueSurfaceView.setVisibility(View.GONE);
-                this.opaqueSurfaceView = null;
-            }
-            if (this.transparentTextureView == null) {
-                this.transparentTextureView = new TextureView(this.getContext());
-                this.transparentTextureView.setLayoutParams(EngineView.childViewLayoutParams);
-                this.transparentTextureView.setSurfaceTextureListener(this);
-                this.transparentTextureView.setOpaque(false);
-                this.addView(this.transparentTextureView);
+            if (topMost) {
+                this.surfaceView = new SurfaceView(this.getContext());
+                this.surfaceView.setLayoutParams(EngineView.childViewLayoutParams);
+                SurfaceHolder surfaceHolder = this.surfaceView.getHolder();
+
+                // ZOrder is not dynamic before Android 11. Recreate the surfaceView and set order before adding to the parent
+                // https://developer.android.com/reference/android/view/SurfaceView#setZOrderOnTop(boolean)
+                this.surfaceView.setZOrderOnTop(true);    // necessary
+                surfaceHolder.setFormat(PixelFormat.TRANSPARENT);
+                this.addView(this.surfaceView);
+            } else if (this.textureView == null) {
```

Comment:
Won't the textureView always be null at this point since it was unconditionally cleared out on lines 88-91?

Reply:
Fixed.